spec/Candid.md

```diff
@@ -1247,6 +1247,9 @@ M(ref(r) : principal) = i8(0)
 M(id(v*) : principal) = i8(1) M(v* : vec nat8)
 ```
 
+Note:
+
+* Since `null`, `reserved`, `record {}`, and records of such values, take no space, to prevent unbounded sized message, we limit the total vector length of such zero-sized values in a message to be 2MiB. For example, if there are both `vec null` and `vec record {}` in a message, the length of both vectors combined cannot exceed 2MiB.
```

Comment:
Maybe it's cleaner to give the length not in bytes, but rather elements? Because the problem is precisely that such elements have different sizes in different encodings.

```suggestion
* Since `null`, `reserved`, `record {}`, and records of such values, take no space, to prevent unbounded sized message, we limit the total vector length of such zero-sized values in a message to be 2465792 elements. For example, if a message contains two vectors, one at type `vec null` and one at type `vec record {}`, then the length of both vectors combined cannot exceed 2465792 elements.
```

Reply:
Sure. Where does the number `2465792` come from? It's more than 2MiB?

Reply:
Did I miscalculate? Indeed I did, fixed :-D

Reply:
Yeah, let's make it 2000000. Easier for the human eyes.